### ISSUE TITLE
How to achieve faster convergence speed?

### ISSUE THREAD

**Issue by z-bookworm** [google-research/prompt-tuning#151]
Hi,

Thank you for the simple yet elegant work!

I wonder if you have encountered any convergence issue? I am finetuning a [XGLM(4.5B)](https://huggingface.co/facebook/xglm-4.5B) for a text generation task, with only 100 tunable prefix embeddings. However, for a training dataset with only one sentence, it takes a long time for model to overfit that sentence. In fact, it never converges :(. I have tries several learning rates from 5e-5 to 5e-3.

Could you give me some suggestions? Thanks in advance.

**Comment by blester125** [comment]
Hi, as you note, we also have noticed that prompt tuning can take more steps than one would like to converge. We think it is because the optimization problem is much harder than when doing finetuning (the gradient is traveling through the whole frozen model). 

The large learning rate was part of our fix for this but I don't think we have found the best hyper parameters yet. I think exploring a learning rate warmup and an even larger learning rate could be a good place to start looking.

Also our followup work [SPoT](https://arxiv.org/abs/2110.07904) (which we are working on adding to the repo) could also help speed things up by starting from a prompt that is already trained to control the model output.

We haven't really dug into the how many parameters you need to learned some tasks, but the intuition is that the model already has a latent ability to solve various different tasks and the prompts are more about guiding it to actually do that task, rather than the prompts being a place to memorize particular examples for a task so I'm not sure that overfitting to an example or the raw parameter count can be extrapolated into the final capabilities of the model.
